src/arkit.cs

```diff
@@ -0,0 +1,493 @@
+//
+// ARKit bindings
+//
+// Authors:
+//	Vincent Dondain  <vidondai@microsoft.com>
+//
+// Copyright 2017 Microsoft Inc. All rights reserved.
+//
+
+// #if XAMCORE_2_0
+
+using System;
+using XamCore.CoreFoundation;
+using XamCore.CoreGraphics;
+using XamCore.CoreVideo;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+using XamCore.SpriteKit;
+using XamCore.SceneKit;
+using XamCore.UIKit;
+
+using Vector3 = global::OpenTK.Vector3;
+using Matrix3 = global::OpenTK.Matrix3;
+using Matrix4 = global::OpenTK.Matrix4;
+
+namespace XamCore.ARKit {
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Native]
+	public enum ARTrackingState : nint
+	{
+		NotAvailable,
+		Limited,
+		Normal,
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Native]
+	public enum ARTrackingStateReason : nint
+	{
+		None,
+		ExcessiveMotion,
+		InsufficientFeatures,
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[ErrorDomain ("ARErrorDomain")]
+	[Native]
+	public enum ARErrorCode : nint
+	{
+		UnsupportedConfiguration = 100,
+		SensorUnavailable = 101,
+		SensorFailed = 102,
+		WorldTrackingFailed = 200,
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Native]
+	public enum ARHitTestResultType : nuint
+	{
+		FeaturePoint = 1 << 0,
+		EstimatedHorizontalPlane = 1 << 1,
+		ExistingPlane = 1 << 3,
+		ExistingPlaneUsingExtent = 1 << 4,
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Native]
+	public enum ARPlaneAnchorAlignment : nint
+	{
+		Horizontal,
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Native]
+	public enum ARSessionRunOptions : nuint
+	{
+		ResetTracking = 1 << 0,
+		RemoveExistingAnchors = 1 << 1,
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Native]
+	public enum ARWorldAlignment : nint
+	{
+		Gravity,
+		GravityAndHeading,
+		Camera,
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Native]
+	public enum ARPlaneDetection : nuint
+	{
+		None = 0,
+		Horizontal = 1 << 0,
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface ARAnchor : NSCopying
+	{
+		[NullAllowed, Export ("identifier")]
+		NSUuid Identifier { get; }
+
+		[Export ("transform")]
+		Matrix4 Transform {
+			[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+			get;
+		}
+
+		[Export ("initWithTransform:")]
+		[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+		IntPtr Constructor (Matrix4 transform);
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface ARCamera : NSCopying
+	{
```

Comment:
Minor, Codig guidelines, should be on the above line. Also this is present in all interfaces